.github/actions/analyze/action.yml

```
@@ -58,6 +61,46 @@ inputs:
 runs:
   using: composite
   steps:
+
+  - name: Get latest commit hash from master
+    id: latest_commit
+    run: |
+      git fetch origin master
+      LATEST_COMMIT=$(git rev-parse origin/master)
+      echo "LATEST_COMMIT=$LATEST_COMMIT" >> $GITHUB_ENV
+    shell: bash
+
+  - name: Check if previous artifact exists
+    id: artifact_check
+    run: |
+      ARTIFACT_EXISTS=$(curl -s -H "Authorization: Bearer ${{ github.token }}" "https://api.github.com/repos/${{ github.repository }}/actions/artifacts?name=reports-${{ env.LATEST_COMMIT }}" | jq '.total_count' | awk '{print $1}')
+      if [[ $ARTIFACT_EXISTS -gt 0 ]]; then
+        echo "ARTIFACT_EXISTS=true" >> $GITHUB_ENV
+      else
+        echo "ARTIFACT_EXISTS=false" >> $GITHUB_ENV
+      fi
+    shell: bash
+
+  - name: Get artifact URL
+    id: get_artifact_url
+    run: |
+      ARTIFACT_URL=$(curl -s -H "Authorization: Bearer ${{ inputs.GITHUB_TOKEN }}" \
```

```suggestion
      ARTIFACT_URL=$(curl -s -H "Authorization: Bearer $GITHUB_TOKEN" \
```

We should define this in an env then use it

---

@markw-deriv security will defiantly complain about this could u please help to update them 🙏 

we should not use `input.` and `env.` inside the steps run

---

This is the auto generated GITHUB_TOKEN from the action. It shouldn't be a PAT or a fixed token in secrets.

---

This is an example of the way we can use it

```
- name: Send custom JSON data to Slack workflow
      env:
        SLACK_WEBHOOK_URL: ${{ inputs.SLACK_WEBHOOK_URL }}
        SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
        MESSAGE: ${{ inputs.MESSAGE }}

      uses: slackapi/slack-github-action@6c661ce58804a1a20f6dc5fbee7f0381b469e001 # v1.25.0
      with:
        # For posting a rich message using Block Kit
        payload: |
          {
            "blocks": [
              {
                "type": "section",
                "text": {
                  "type": "mrkdwn",
                  "text": ${{ toJSON(env.MESSAGE) }}
                }
              }
            ]
          }
```